src/library_dylink.js

```diff
@@ -311,7 +311,12 @@ var LibraryDylink = {
       end = binary.length
     } else {
       var int32View = new Uint32Array(new Uint8Array(binary.subarray(0, 24)).buffer);
-      failIf(int32View[0] != 0x6d736100, 'need to see wasm magic number'); // \0asm
+#if SUPPORT_BIG_ENDIAN
+      var magicNumber = 0x0061736d;
+#else
+      var magicNumber = 0x6d736100;
+#endif
+      failIf(int32View[0] != magicNumber, 'need to see wasm magic number'); // \0asm
```

Comment:
It looks like this change would make the output only work on big endian with that option. But I think the idea is to support it, not require it. For comparison, here is another check:

https://github.com/emscripten-core/emscripten/blob/4a7cfe26f0915adc464e2eb10630e332f0d1b663/src/runtime_assertions.js#L9-L17

That code just skips the check if we support big-endian.

For the code here, how about asserting that it is one of the two possible values if big-endian is supported?

Reply:
Thanks for reviewing. I have changed it to check for both orders when BE is supported.